[SecuritySolution] Comma-separate array values in the alerts table

janmonschke opened the pull request:
## Summary

As described in https://github.com/elastic/kibana/issues/125770, looking at array values in the alerts table is confusing because individual values are not properly separated. This PR implements [the request from design](https://github.com/elastic/kibana/issues/125770#issuecomment-1097027002) to comma-separate array values in the table.

### Steps to reproduce

- Make sure you have alerts with process information generated
- Go to the alerts page
- Filter for `process.args : *`
- Find a process with multiple arguments and hover the cell to check that the values are comma-separated

https://user-images.githubusercontent.com/68591/171405413-11362f7c-60c8-4d8a-a890-2a3ed3b65223.mov

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

janmonschke committed:
fix: comma-separate array values

janmonschke committed:
test: update snapshots

janmonschke committed:
fix: make sure that individual values are still draggable

janmonschke commented:
I noticed the initial changes caused a regression with draggable fields in timeline:

<img width="852" alt="Screenshot 2022-06-02 at 18 04 06" src="https://user-images.githubusercontent.com/68591/171674800-d3257513-7174-4c5f-af3f-34588a970d33.png">

<hr/>

**In timeline, individual fields should be draggable. The changes in https://github.com/elastic/kibana/pull/133297/commits/16483e695c330e18e5a535eb27838ed1f68f86c4 make sure this behaviour is retained:**

<hr/>

<img width="853" alt="Screenshot 2022-06-02 at 18 05 47" src="https://user-images.githubusercontent.com/68591/171674814-c03f32e8-cf98-4123-8c25-568160b50dd8.png">

janmonschke committed:
chore: reintroduce `parseValue`

We currently don't know exactly if it's still required or not. Hence we're reintroducing it and will revisit its usage in a future PR.

janmonschke commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into security/comma-separate-array-values

michaelolo24 commented on x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/plain_column_renderer.tsx:
```diff
@@ -43,23 +43,60 @@ export const plainColumnRenderer: ColumnRenderer = {
+      return values[0];
+    }
+  }
+  return undefined;
```
per our offline convo, maybe we can be more defensive here in case values happens to be something else, but I didn't see anything strange in testing

michaelolo24 reviewed (approved):
thanks! LGTM 🚀

janmonschke committed:
chore: preserve `value`'s value

janmonschke committed:
chore: remove `parseValue`

`parseValue` was introduced in https://github.com/elastic/kibana/commit/de6958d96feb22737b20ed24e2743bffd16d06fe when the type of `value` included `object`. The type has been narrowed since then and does not include `object` anymore. Hence we are removing this runtime check.

janmonschke commented:
I removed `parseValue` in https://github.com/elastic/kibana/pull/133297/commits/c243ca7aacbc076e82a0b012d1526160c9fe9673. I explained the reasoning in the commit message but will copy it here as well, to make it easier to find:

> `parseValue` was introduced in https://github.com/elastic/kibana/commit/de6958d96feb22737b20ed24e2743bffd16d06fe when the type of `value` included `object`. The type has been narrowed since then and does not include `object` anymore. Hence we are removing this runtime check.

angorayc commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into security/comma-separate-array-values

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.3|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/133617">](https://github.com/elastic/kibana/pull/133617)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)
<!--- Backport version: 7.4.0 -->

karanbirsingh-qasource commented:
Hi @michaelolo24 

validated issue by testing pr locally and there issue is fixed ✔️ . we will going to check this issue on BC4 as this commits are not there on BC3 of 8.3.0.

**Build Details:**
`branch: security/comma-separate-array-values`


**Screen-Shoot:**

![image](https://user-images.githubusercontent.com/59917825/172595320-019eaadc-b133-4736-b652-590b65ca7fac.png)


thanks !!
